DialogflowDemo/DialogflowDemo/Controllers/MentorSearchController.cs

```diff
@@ -0,0 +1,144 @@
+        }
+
+        [HttpGet, HttpPost]
+        public async Task<ContentResult> FulfilRequest([FromBody] JObject rawRequest)
+        {
+            var queryResult = (JObject) rawRequest["queryResult"];
+            queryResult.Remove("fulfillmentMessages");
+
+            if (queryResult.TryGetValue("sentimentAnalysisResult", out var sentimentToken) && sentimentToken is JObject sentimentObject &&
+                sentimentObject.TryGetValue("queryTextSentiment", out var querySentimentToken) && querySentimentToken is JObject querySentimentObject &&
+                querySentimentObject.TryGetValue("score", out var scoreToken) &&
+                querySentimentObject.TryGetValue("magnitude", out var magnitudeToken))
+            {
+                double score = (double) scoreToken;
+                double magnitude = (double) magnitudeToken;
+                if (score < 0 && magnitude > 0.5)
+                {
+                    var transfer = new WebhookResponse { FollowupEventInput = new EventInput { Name = "telephone-event" } };
+                    return Content(transfer.ToString(), "application/json");
+                }
+            }
+
+            // Parse the body of the request using the Protobuf JSON parser,
+            // *not* Json.NET.
+            WebhookRequest request;
+            using (var reader = new StringReader(rawRequest.ToString()))
+            {
+                request = jsonParser.Parse<WebhookRequest>(reader);
+            }
+
+            // TODO: Authenticate the request
+
+
+            WebhookResponse response = await GetResponseAsync(request);
+            // Ask Protobuf to format the JSON to return.
+            // Again, we don't want to use Json.NET - it doesn't know how to handle Struct
+            // values etc.
+            string responseJson = response.ToString();
+            return Content(responseJson, "application/json");
+        }
+
+        private async Task<WebhookResponse> GetResponseAsync(WebhookRequest request)
+        {
+            
+            switch (request.QueryResult.Intent.DisplayName)
+            {
+                case "Default Welcome Intent":
+                    return GetWelcomeResponse(request);
+                case "Default Fallback Intent":
+                    return GetFallbackResponse(request);
+                case "Mentor Search":
+                    return await GetMentorSearchResponse(request);
+                case "Transfer Call Escalation to Human":
+                    return GetTransferCallResponse(request);
+                case "Goodbye":
+                    return GetGoodbyeResponse(request);
+                default:
+                    return new WebhookResponse { FulfillmentText = "Sorry, I didn't understand that intent." };
+            }
+        }
+
+        private static WebhookResponse GetWelcomeResponse(WebhookRequest request) =>
+            CreateResponse("Greetings! What mentor are you looking for?");
```

Comment:
CreateResponse("Greetings! What mentor are you looking for? You can search by name, city or topic.");